hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/services/AbfsClient.java

```diff
@@ -217,8 +220,29 @@ protected AbfsPerfTracker getAbfsPerfTracker() {
     return abfsPerfTracker;
   }
 
-  ExponentialRetryPolicy getRetryPolicy() {
-    return retryPolicy;
+  ExponentialRetryPolicy getExponentialRetryPolicy() {
+    return exponentialRetryPolicy;
+  }
+
+  StaticRetryPolicy getStaticRetryPolicy() {
+    return staticRetryPolicy;
+  }
+
+  /**
+   * Returns the retry policy to be used for Abfs Rest Operation Failure
+   * @param failureReason helps to decide which type of retryPolicy to be used
+   * @return retry policy to be used
+   */
+  public AbfsRetryPolicy getRetryPolicy(final String failureReason) {
+    if (!CONNECTION_TIMEOUT_ABBREVIATION.equals(failureReason)) {
+      return getExponentialRetryPolicy();
+    }
+
+    if (getAbfsConfiguration().getStaticRetryForConnectionTimeoutEnabled()) {
+      return getStaticRetryPolicy();
+    } else {
+      return getExponentialRetryPolicy();
+    }
```

Comment:
public AbfsRetryPolicy getRetryPolicy(final String failureReason) {
    return CONNECTION_TIMEOUT_ABBREVIATION.equals(failureReason) && getAbfsConfiguration().getStaticRetryForConnectionTimeoutEnabled() ? getStaticRetryPolicy() : getExponentialRetryPolicy();
  } if else can be simplified

Reply:
Taken